src/main/java/taxi/service/PasswordUtilImpl.java

```diff
@@ -0,0 +1,76 @@
+package taxi.service;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.KeySpec;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import taxi.lib.Service;
+
+@Service
+public class PasswordUtilImpl implements PasswordUtil {
+    private static final int ITERATIONS = 65536;
+    private static final int KEY_LENGTH = 128;
+    private static final String ALGORITHM = "PBKDF2WithHmacSHA1";
+
+    @Override
+    public String generateHash(String password) {
+        try {
+            byte[] salt = generateSalt();
+            byte[] hash = getHash(password, salt);
+            return toHex(salt) + ":" + toHex(hash);
+        } catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public boolean checkPassword(String hashFromDb, String password) {
```

Comment:
In this method:
1. Hash loginPassword
2. check it with dbPassword using equals()

Delete other extra staff